include/mesh.tcc

```diff
@@ -299,22 +298,84 @@ void mpm::Mesh<Tdim>::compute_cell_neighbours() {
   }
 
   // Assign neighbour to cells
-  tbb::parallel_for(
-      tbb::blocked_range<int>(size_t(0), size_t(cells_.size()),
-                              tbb_grain_size_),
-      [&](const tbb::blocked_range<int>& range) {
-        for (int i = range.begin(); i != range.end(); ++i) {
-          // Iterate over each node in current cell
-          for (auto id : cells_[i]->nodes_id()) {
-            auto cell_id = cells_[i]->id();
-            // Get the cells associated with each node
-            for (auto neighbour_id : node_cell_map[id])
-              if (neighbour_id != cell_id)
-                cells_[i]->add_neighbour(neighbour_id);
-          }
-        }
-      },
-      tbb::simple_partitioner());
+  tbb::parallel_for(tbb::blocked_range<int>(size_t(0), size_t(cells_.size()),
+                                            tbb_grain_size_),
+                    [&](const tbb::blocked_range<int>& range) {
+                      for (int i = range.begin(); i != range.end(); ++i) {
+                        // Iterate over each node in current cell
+                        for (auto id : cells_[i]->nodes_id()) {
+                          auto cell_id = cells_[i]->id();
+                          // Get the cells associated with each node
+                          for (auto neighbour_id : node_cell_map[id])
+                            if (neighbour_id != cell_id)
+                              cells_[i]->add_neighbour(neighbour_id);
+                        }
+                      }
+                    },
+                    tbb::simple_partitioner());
+}
+
+//! Find particle neighbours for all particle
+template <unsigned Tdim>
+void mpm::Mesh<Tdim>::compute_particle_neighbours(const Cell<Tdim>& cell) {
+  // Check argument passed
+  Container<Cell<Tdim>> cell_ctr;
+  cell_ctr.add(cell);
+  auto& cells = (cell_ctr == nullptr) ? cells_ : cell_ctr;
+
+  // Check whether cells have been assigned with neighbours
+  if ((*cells_.cbegin())->nneighbours() == 0)
+    throw std::runtime_error(
+        "No neighbours have been assigned to cell, cannot "
+        "compute_particle_neighbours for single cell");
+
+  // Loop over cells
+  for (auto citr = cells.cbegin(); citr != cells.cend(); ++citr) {
+    // Loop over the neighbouring cell particles
```

Comment:
This comment is not very clear on what the following line or the iteration is doing